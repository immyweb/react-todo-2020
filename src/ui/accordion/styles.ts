import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
