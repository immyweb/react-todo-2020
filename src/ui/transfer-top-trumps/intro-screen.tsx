import * as React from 'react';

import { Gradient } from './gradient';
import ButtonCta from './button-cta';

import {
  TTLozengeBtn,
  BadgeImageHolder,
  Badge,
  Inner,
  TTHeadline,
  TTSubHeadline,
  IntroHeader,
  IntroInstructions,
  IntroContent,
  IntroInstructionsContent,
  IntroInstructionsContentP,
  BtnHolder,
  Divider,
  BadgeImage,
} from './styles';
import { colours } from '../../styles/colours';

import { IIntro, IClub } from './types';
import { ITheme } from '../../types/theme';

interface IProps extends IIntro {
  className?: string;
  theme: ITheme;
  targetClub: IClub;
  onStartGame(): void;
}

export const IntroScreen: React.SFC<IProps> = ({
  headline,
  subHeadline,
  introHeadline,
  introCopy,
  no,
  yes,
  playBtn,
  theme,
  targetClub,
  onStartGame,
}) => {
  return (
    <div>
      <Inner className={'ttt-intro'}>
        <TTHeadline className={'ttt-intro-headline'} fontColour={theme.primary}>
          {headline}
        </TTHeadline>
        <TTSubHeadline className={'ttt-intro-subheadline'} fontColour={colours.$white}>
          {subHeadline}
        </TTSubHeadline>
        <BadgeImageHolder className={'ttt-intro-badge'}>
          <Gradient colour={targetClub.teamColours} top={30} />
          <Badge size={85}>
            <BadgeImage src={targetClub.badge} alt={targetClub.name} />
          </Badge>
        </BadgeImageHolder>
        <IntroHeader className={'ttt-intro-header'}>{introHeadline}</IntroHeader>
        <IntroContent className={'ttt-intro-content'} color={theme.primary}>
          {introCopy}
        </IntroContent>
        <IntroInstructions className={'ttt-intro-instructions'}>
          <IntroInstructionsContent reverse={true}>
            <ButtonCta
              bkgndTop={colours.$lightRed}
              bkgndBottom={colours.$darkRed}
              iconName={'cross-solid'}
              disabled={true}
            />
            <IntroInstructionsContentP>{no}</IntroInstructionsContentP>
          </IntroInstructionsContent>
          <Divider />
          <IntroInstructionsContent>
            <ButtonCta
              bkgndTop={colours.$lightGreen}
              bkgndBottom={colours.$darkGreen}
              iconName={'tick-solid'}
              disabled={true}
            />
            <IntroInstructionsContentP>{yes}</IntroInstructionsContentP>
          </IntroInstructionsContent>
        </IntroInstructions>
        <BtnHolder className={'ttt-intro-button'}>
          <TTLozengeBtn
            bkgndTop={theme.primary}
            bkgndBottom={theme.secondary}
            onClick={onStartGame}
          >
            {playBtn}
          </TTLozengeBtn>
        </BtnHolder>
      </Inner>
    </div>
  );
};
