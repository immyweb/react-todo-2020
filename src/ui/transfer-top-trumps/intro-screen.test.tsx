import * as React from 'react';
import { shallow } from 'enzyme';

import { IntroScreen } from './intro-screen';

import { TTLozengeBtn, TTHeadline, IntroContent } from './styles';

const targetClub = {
  id: 'man-utd',
  name: 'Man Utd',
  badge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png',
  teamColours: '#a81507',
};

function shallowRender(props?: any) {
  return shallow(
    <IntroScreen
      headline={'Transfer Top Trumps'}
      subHeadline={'Who will you choose?'}
      introHeadline={'How to play'}
      introCopy={'Choose the players you would like your club to sign'}
      no={`Tap/Click for 'No'`}
      yes={`Tap/Click for 'Yes'`}
      playBtn={'Play now'}
      theme={{ primary: '#000000', secondary: '#ffffff' }}
      targetClub={targetClub}
      onStartGame={() => {}}
      {...props}
    />,
  );
}

describe('<IntroScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onStartGame when play button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onStartGame: fn });
    const btn = wrapper.find(TTLozengeBtn).last();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('headings are theme-able', () => {
    const wrapper = shallowRender({
      theme: { primary: '#e665bf', secondary: '#a6498a' },
    });
    const headline = wrapper.find(TTHeadline);
    const intro = wrapper.find(IntroContent);
    expect(headline.props().fontColour).toEqual('#e665bf');
    expect(intro.props().color).toEqual('#e665bf');
  });
});
