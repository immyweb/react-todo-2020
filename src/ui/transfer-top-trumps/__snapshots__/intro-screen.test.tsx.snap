// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    headline="Transfer Top Trumps"
    introCopy="Choose the players you would like your club to sign"
    introHeadline="How to play"
    no="Tap/Click for 'No'"
    onStartGame={[Function]}
    playBtn="Play now"
    subHeadline="Who will you choose?"
    targetClub={
      Object {
        "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
        "id": "man-utd",
        "name": "Man Utd",
        "teamColours": "#a81507",
      }
    }
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
    yes="Tap/Click for 'Yes'"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <glamorous(glamorous.Div)
        className="ttt-intro"
      >
        <glamorous(glamorous(h3))
          className="ttt-intro-headline"
          fontColour="#000000"
        >
          Transfer Top Trumps
        </glamorous(glamorous(h3))>
        <glamorous(glamorous(h2))
          className="ttt-intro-subheadline"
          fontColour="#ffffff"
        >
          Who will you choose?
        </glamorous(glamorous(h2))>
        <glamorous(div)
          className="ttt-intro-badge"
        >
          <Unknown
            colour="#a81507"
            top={30}
          />
          <glamorous(div)
            size={85}
          >
            <glamorous(img)
              alt="Man Utd"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
            />
          </glamorous(div)>
        </glamorous(div)>
        <glamorous(glamorous(h3))
          className="ttt-intro-header"
        >
          How to play
        </glamorous(glamorous(h3))>
        <glamorous(glamorous(glamorous.P))
          className="ttt-intro-content"
          color="#000000"
        >
          Choose the players you would like your club to sign
        </glamorous(glamorous(glamorous.P))>
        <glamorous(div)
          className="ttt-intro-instructions"
        >
          <glamorous(div)
            reverse={true}
          >
            <ButtonCta
              bkgndBottom="#9f041b"
              bkgndTop="#ed283a"
              disabled={true}
              iconName="cross-solid"
            />
            <glamorous(glamorous(glamorous.P))>
              Tap/Click for 'No'
            </glamorous(glamorous(glamorous.P))>
          </glamorous(div)>
          <glamorous(div) />
          <glamorous(div)>
            <ButtonCta
              bkgndBottom="#6cb71b"
              bkgndTop="#a5dc39"
              disabled={true}
              iconName="tick-solid"
            />
            <glamorous(glamorous(glamorous.P))>
              Tap/Click for 'Yes'
            </glamorous(glamorous(glamorous.P))>
          </glamorous(div)>
        </glamorous(div)>
        <glamorous(div)
          className="ttt-intro-button"
        >
          <glamorous(glamorous(button))
            bkgndBottom="#ffffff"
            bkgndTop="#000000"
            onClick={[Function]}
          >
            Play now
          </glamorous(glamorous(button))>
        </glamorous(div)>
      </glamorous(glamorous.Div)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(glamorous(h3))
            className="ttt-intro-headline"
            fontColour="#000000"
          >
            Transfer Top Trumps
          </glamorous(glamorous(h3))>,
          <glamorous(glamorous(h2))
            className="ttt-intro-subheadline"
            fontColour="#ffffff"
          >
            Who will you choose?
          </glamorous(glamorous(h2))>,
          <glamorous(div)
            className="ttt-intro-badge"
          >
            <Unknown
              colour="#a81507"
              top={30}
            />
            <glamorous(div)
              size={85}
            >
              <glamorous(img)
                alt="Man Utd"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
              />
            </glamorous(div)>
          </glamorous(div)>,
          <glamorous(glamorous(h3))
            className="ttt-intro-header"
          >
            How to play
          </glamorous(glamorous(h3))>,
          <glamorous(glamorous(glamorous.P))
            className="ttt-intro-content"
            color="#000000"
          >
            Choose the players you would like your club to sign
          </glamorous(glamorous(glamorous.P))>,
          <glamorous(div)
            className="ttt-intro-instructions"
          >
            <glamorous(div)
              reverse={true}
            >
              <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                disabled={true}
                iconName="cross-solid"
              />
              <glamorous(glamorous(glamorous.P))>
                Tap/Click for 'No'
              </glamorous(glamorous(glamorous.P))>
            </glamorous(div)>
            <glamorous(div) />
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#6cb71b"
                bkgndTop="#a5dc39"
                disabled={true}
                iconName="tick-solid"
              />
              <glamorous(glamorous(glamorous.P))>
                Tap/Click for 'Yes'
              </glamorous(glamorous(glamorous.P))>
            </glamorous(div)>
          </glamorous(div)>,
          <glamorous(div)
            className="ttt-intro-button"
          >
            <glamorous(glamorous(button))
              bkgndBottom="#ffffff"
              bkgndTop="#000000"
              onClick={[Function]}
            >
              Play now
            </glamorous(glamorous(button))>
          </glamorous(div)>,
        ],
        "className": "ttt-intro",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Transfer Top Trumps",
            "className": "ttt-intro-headline",
            "fontColour": "#000000",
          },
          "ref": null,
          "rendered": "Transfer Top Trumps",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Who will you choose?",
            "className": "ttt-intro-subheadline",
            "fontColour": "#ffffff",
          },
          "ref": null,
          "rendered": "Who will you choose?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown
                colour="#a81507"
                top={30}
              />,
              <glamorous(div)
                size={85}
              >
                <glamorous(img)
                  alt="Man Utd"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
                />
              </glamorous(div)>,
            ],
            "className": "ttt-intro-badge",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "colour": "#a81507",
                "top": 30,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <glamorous(img)
                  alt="Man Utd"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
                />,
                "size": 85,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "Man Utd",
                  "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "How to play",
            "className": "ttt-intro-header",
          },
          "ref": null,
          "rendered": "How to play",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Choose the players you would like your club to sign",
            "className": "ttt-intro-content",
            "color": "#000000",
          },
          "ref": null,
          "rendered": "Choose the players you would like your club to sign",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                reverse={true}
              >
                <ButtonCta
                  bkgndBottom="#9f041b"
                  bkgndTop="#ed283a"
                  disabled={true}
                  iconName="cross-solid"
                />
                <glamorous(glamorous(glamorous.P))>
                  Tap/Click for 'No'
                </glamorous(glamorous(glamorous.P))>
              </glamorous(div)>,
              <glamorous(div) />,
              <glamorous(div)>
                <ButtonCta
                  bkgndBottom="#6cb71b"
                  bkgndTop="#a5dc39"
                  disabled={true}
                  iconName="tick-solid"
                />
                <glamorous(glamorous(glamorous.P))>
                  Tap/Click for 'Yes'
                </glamorous(glamorous(glamorous.P))>
              </glamorous(div)>,
            ],
            "className": "ttt-intro-instructions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <ButtonCta
                    bkgndBottom="#9f041b"
                    bkgndTop="#ed283a"
                    disabled={true}
                    iconName="cross-solid"
                  />,
                  <glamorous(glamorous(glamorous.P))>
                    Tap/Click for 'No'
                  </glamorous(glamorous(glamorous.P))>,
                ],
                "reverse": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bkgndBottom": "#9f041b",
                    "bkgndTop": "#ed283a",
                    "disabled": true,
                    "iconName": "cross-solid",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Tap/Click for 'No'",
                  },
                  "ref": null,
                  "rendered": "Tap/Click for 'No'",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <ButtonCta
                    bkgndBottom="#6cb71b"
                    bkgndTop="#a5dc39"
                    disabled={true}
                    iconName="tick-solid"
                  />,
                  <glamorous(glamorous(glamorous.P))>
                    Tap/Click for 'Yes'
                  </glamorous(glamorous(glamorous.P))>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bkgndBottom": "#6cb71b",
                    "bkgndTop": "#a5dc39",
                    "disabled": true,
                    "iconName": "tick-solid",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Tap/Click for 'Yes'",
                  },
                  "ref": null,
                  "rendered": "Tap/Click for 'Yes'",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <glamorous(glamorous(button))
              bkgndBottom="#ffffff"
              bkgndTop="#000000"
              onClick={[Function]}
            >
              Play now
            </glamorous(glamorous(button))>,
            "className": "ttt-intro-button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bkgndBottom": "#ffffff",
              "bkgndTop": "#000000",
              "children": "Play now",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Play now",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <glamorous(glamorous.Div)
          className="ttt-intro"
        >
          <glamorous(glamorous(h3))
            className="ttt-intro-headline"
            fontColour="#000000"
          >
            Transfer Top Trumps
          </glamorous(glamorous(h3))>
          <glamorous(glamorous(h2))
            className="ttt-intro-subheadline"
            fontColour="#ffffff"
          >
            Who will you choose?
          </glamorous(glamorous(h2))>
          <glamorous(div)
            className="ttt-intro-badge"
          >
            <Unknown
              colour="#a81507"
              top={30}
            />
            <glamorous(div)
              size={85}
            >
              <glamorous(img)
                alt="Man Utd"
                src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
              />
            </glamorous(div)>
          </glamorous(div)>
          <glamorous(glamorous(h3))
            className="ttt-intro-header"
          >
            How to play
          </glamorous(glamorous(h3))>
          <glamorous(glamorous(glamorous.P))
            className="ttt-intro-content"
            color="#000000"
          >
            Choose the players you would like your club to sign
          </glamorous(glamorous(glamorous.P))>
          <glamorous(div)
            className="ttt-intro-instructions"
          >
            <glamorous(div)
              reverse={true}
            >
              <ButtonCta
                bkgndBottom="#9f041b"
                bkgndTop="#ed283a"
                disabled={true}
                iconName="cross-solid"
              />
              <glamorous(glamorous(glamorous.P))>
                Tap/Click for 'No'
              </glamorous(glamorous(glamorous.P))>
            </glamorous(div)>
            <glamorous(div) />
            <glamorous(div)>
              <ButtonCta
                bkgndBottom="#6cb71b"
                bkgndTop="#a5dc39"
                disabled={true}
                iconName="tick-solid"
              />
              <glamorous(glamorous(glamorous.P))>
                Tap/Click for 'Yes'
              </glamorous(glamorous(glamorous.P))>
            </glamorous(div)>
          </glamorous(div)>
          <glamorous(div)
            className="ttt-intro-button"
          >
            <glamorous(glamorous(button))
              bkgndBottom="#ffffff"
              bkgndTop="#000000"
              onClick={[Function]}
            >
              Play now
            </glamorous(glamorous(button))>
          </glamorous(div)>
        </glamorous(glamorous.Div)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(h3))
              className="ttt-intro-headline"
              fontColour="#000000"
            >
              Transfer Top Trumps
            </glamorous(glamorous(h3))>,
            <glamorous(glamorous(h2))
              className="ttt-intro-subheadline"
              fontColour="#ffffff"
            >
              Who will you choose?
            </glamorous(glamorous(h2))>,
            <glamorous(div)
              className="ttt-intro-badge"
            >
              <Unknown
                colour="#a81507"
                top={30}
              />
              <glamorous(div)
                size={85}
              >
                <glamorous(img)
                  alt="Man Utd"
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
                />
              </glamorous(div)>
            </glamorous(div)>,
            <glamorous(glamorous(h3))
              className="ttt-intro-header"
            >
              How to play
            </glamorous(glamorous(h3))>,
            <glamorous(glamorous(glamorous.P))
              className="ttt-intro-content"
              color="#000000"
            >
              Choose the players you would like your club to sign
            </glamorous(glamorous(glamorous.P))>,
            <glamorous(div)
              className="ttt-intro-instructions"
            >
              <glamorous(div)
                reverse={true}
              >
                <ButtonCta
                  bkgndBottom="#9f041b"
                  bkgndTop="#ed283a"
                  disabled={true}
                  iconName="cross-solid"
                />
                <glamorous(glamorous(glamorous.P))>
                  Tap/Click for 'No'
                </glamorous(glamorous(glamorous.P))>
              </glamorous(div)>
              <glamorous(div) />
              <glamorous(div)>
                <ButtonCta
                  bkgndBottom="#6cb71b"
                  bkgndTop="#a5dc39"
                  disabled={true}
                  iconName="tick-solid"
                />
                <glamorous(glamorous(glamorous.P))>
                  Tap/Click for 'Yes'
                </glamorous(glamorous(glamorous.P))>
              </glamorous(div)>
            </glamorous(div)>,
            <glamorous(div)
              className="ttt-intro-button"
            >
              <glamorous(glamorous(button))
                bkgndBottom="#ffffff"
                bkgndTop="#000000"
                onClick={[Function]}
              >
                Play now
              </glamorous(glamorous(button))>
            </glamorous(div)>,
          ],
          "className": "ttt-intro",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Transfer Top Trumps",
              "className": "ttt-intro-headline",
              "fontColour": "#000000",
            },
            "ref": null,
            "rendered": "Transfer Top Trumps",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Who will you choose?",
              "className": "ttt-intro-subheadline",
              "fontColour": "#ffffff",
            },
            "ref": null,
            "rendered": "Who will you choose?",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Unknown
                  colour="#a81507"
                  top={30}
                />,
                <glamorous(div)
                  size={85}
                >
                  <glamorous(img)
                    alt="Man Utd"
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
                  />
                </glamorous(div)>,
              ],
              "className": "ttt-intro-badge",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "colour": "#a81507",
                  "top": 30,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <glamorous(img)
                    alt="Man Utd"
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png"
                  />,
                  "size": 85,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "alt": "Man Utd",
                    "src": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "How to play",
              "className": "ttt-intro-header",
            },
            "ref": null,
            "rendered": "How to play",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Choose the players you would like your club to sign",
              "className": "ttt-intro-content",
              "color": "#000000",
            },
            "ref": null,
            "rendered": "Choose the players you would like your club to sign",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <glamorous(div)
                  reverse={true}
                >
                  <ButtonCta
                    bkgndBottom="#9f041b"
                    bkgndTop="#ed283a"
                    disabled={true}
                    iconName="cross-solid"
                  />
                  <glamorous(glamorous(glamorous.P))>
                    Tap/Click for 'No'
                  </glamorous(glamorous(glamorous.P))>
                </glamorous(div)>,
                <glamorous(div) />,
                <glamorous(div)>
                  <ButtonCta
                    bkgndBottom="#6cb71b"
                    bkgndTop="#a5dc39"
                    disabled={true}
                    iconName="tick-solid"
                  />
                  <glamorous(glamorous(glamorous.P))>
                    Tap/Click for 'Yes'
                  </glamorous(glamorous(glamorous.P))>
                </glamorous(div)>,
              ],
              "className": "ttt-intro-instructions",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <ButtonCta
                      bkgndBottom="#9f041b"
                      bkgndTop="#ed283a"
                      disabled={true}
                      iconName="cross-solid"
                    />,
                    <glamorous(glamorous(glamorous.P))>
                      Tap/Click for 'No'
                    </glamorous(glamorous(glamorous.P))>,
                  ],
                  "reverse": true,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "bkgndBottom": "#9f041b",
                      "bkgndTop": "#ed283a",
                      "disabled": true,
                      "iconName": "cross-solid",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Tap/Click for 'No'",
                    },
                    "ref": null,
                    "rendered": "Tap/Click for 'No'",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <ButtonCta
                      bkgndBottom="#6cb71b"
                      bkgndTop="#a5dc39"
                      disabled={true}
                      iconName="tick-solid"
                    />,
                    <glamorous(glamorous(glamorous.P))>
                      Tap/Click for 'Yes'
                    </glamorous(glamorous(glamorous.P))>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "bkgndBottom": "#6cb71b",
                      "bkgndTop": "#a5dc39",
                      "disabled": true,
                      "iconName": "tick-solid",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Tap/Click for 'Yes'",
                    },
                    "ref": null,
                    "rendered": "Tap/Click for 'Yes'",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <glamorous(glamorous(button))
                bkgndBottom="#ffffff"
                bkgndTop="#000000"
                onClick={[Function]}
              >
                Play now
              </glamorous(glamorous(button))>,
              "className": "ttt-intro-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bkgndBottom": "#ffffff",
                "bkgndTop": "#000000",
                "children": "Play now",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Play now",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
